chainer/distribution.py

```diff
@@ -0,0 +1,260 @@
+            ~chainer.Variable: Output variable representing Inverse Cumulative
+            Distribution Function.
+
+        """
+        raise NotImplementedError
+
+    def log_cdf(self, x):
+        """Returns logarithm of Cumulative Distribution Function for a input Variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing logarithm of
+            Cumulative Distribution Function.
+
+        """
+        raise NotImplementedError
+
+    def log_prob(self, x):
+        """Returns logarithm of probability for a input variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing logarithm of
+            probability.
+
+        """
+        raise NotImplementedError
+
+    def log_survival_function(self, x):
+        """Returns logarithm of survival function for a input Variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing logarithm of
+            survival function for a input variable.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def mean(self):
+        """Returns mean value.
+
+        Returns:
+            ~chainer.Variable: Output variable representing mean value.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def mode(self):
+        """Returns mode.
+
+        Returns:
+            ~chainer.Variable: Output variable representing mode.
+
+        """
+        raise NotImplementedError
+
+    def perplexity(self, x):
+        """Returns perplexity function for a input variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing perplexity function
+            for a input variable.
+
+        """
+        raise NotImplementedError
+
+    def prob(self, x):
+        """Returns probability for a input variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing probability.
+
+        """
+        raise NotImplementedError
+
+    def sample(self, shape=()):
+        """Samples from this distribution.
```

Comment:
Be more explicit: samples what? My suggestion is "Samples[Draws] random points from the distribution".